### ISSUE TITLE
tedious mssql connect error

### ISSUE THREAD

**Issue by jelhub** [oven-sh/bun#10215]
### What version of Bun is running?

1.1.3+2615dc742

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

Running latest version of bun for windows (v1.1.3)

Using package tedious for mssql connect and queries.

Tedious connects and runs fine using nodejs, but unable to connect when using bun.
Please also see this issue in relation to #5147

**How to reproduce connect error:**
```
bun init -y
bun add tedious
```

**index.js:**
```
import { Connection } from 'tedious'

const connection = new Connection(
  {
    server: 'MyServer',
    authentication: {
      type: 'default',
      options: { userName: 'sa', password: 'password' }
    },
    options: {
      instanceName: '',
      port: 1433,
      database: 'MyDatabase',
      useColumnNames: true,
      rowCollectionOnRequestCompletion: true,
      encrypt: false
    }
  }
)

connection.on('connect', function (err) {
  if (err) {
    console.log('connect error: ', err)
    return
  }
  console.log('connect OK')
  connection.close()
})

// Initialize the connection.
connection.connect()
```

**Test using nodejs**
> node index.js
```connect OK```

**Test using bun (hanging for a while before any output):**
> bun index.js
```
connect error:  4 |   value: true
5 | });
6 | exports.RequestError = exports.ConnectionError = void 0;
7 | class ConnectionError extends Error {
8 |   constructor(message, code) {
9 |     super(message);
        ^
error: Failed to connect to MyServer:1433 in 15000ms
 code: "ETIMEOUT"

      at new ConnectionError (C:\temp\xx\node_modules\tedious\lib\errors.js:9:5)
      at connectTimeout (C:\temp\xx\node_modules\tedious\lib\connection.js:1207:26)
      at C:\temp\xx\node_modules\tedious\lib\connection.js:1152:7

4 |   value: true
5 | });
6 | exports.RequestError = exports.ConnectionError = void 0;
7 | class ConnectionError extends Error {
8 |   constructor(message, code) {
9 |     super(message);
        ^
error: Connection lost - unexpected end of message stream
 code: "ESOCKET"

      at new ConnectionError (C:\temp\xx\node_modules\tedious\lib\errors.js:9:5)
      at socketError (C:\temp\xx\node_modules\tedious\lib\connection.js:1342:26)
      at C:\temp\xx\node_modules\tedious\lib\connection.js:2217:18
```

**Comment by jelhub** [comment]
FYI,
Fails on Windows
Connect OK on Linux

**Comment by jelhub** [comment]
Problem fixed in version 1.1.9
